mgmtsystem/azure.py

```diff
@@ -193,23 +195,85 @@ def capture_vm(self, vm_name, resource_group, container, image_name):
+                foreach ($stack_item in $Stack){{
+                    $stack_name = $stack_item.ToString()
+                    $list.add($stack_name, @{{}})
+                    $res_list = (Get-AzureRmResourceGroupDeploymentOperation -DeploymentName `
+                    $stack_name -ResourceGroupName \"{rg}\").Properties.TargetResource.id
+                    foreach ($res_type in $res_list){{
+                        if ($res_type -like "*virtualmachine*"){{
+                            if(-not $list.item($stack_name).ContainsKey("vms")){{
+                                $list.item($stack_name).add("vms",@{{}})}}
+                            $vm_name = $res_type -split '/' | Select-Object -Last 1
+                            $cur_vm = Get-AzureRmVM -ResourceGroupName \"{rg}\" -Name $vm_name `
+                            -ErrorAction SilentlyContinue
+                            if(!$cur_cur){{$vm_exists = "false"}}
+                            else{{$vm_exists = "true"}}
+                            $list.Item($stack_name).item("vms").add($vm_name, $vm_exists)
+                            Clear-Variable -Name cur_vm
+                        }}
+                        elseif ($res_type -like "*networkinterface*"){{
+                            if(-not $list.item($stack_name).ContainsKey("nics")){{
+                                $list.item($stack_name).add("nics",@{{}})}}
+                            $nic_name = $res_type -split '/' | Select-Object -Last 1
+                            $cur_nic = Get-AzureRmNetworkInterface -Name $nic_name `
+                            -ResourceGroupName \"{rg}\" -ErrorAction SilentlyContinue
+                            if(!$cur_nic){{$nic_exists = "false"}}
+                            else{{$nic_exists = "true"}}
+                            $list.Item($stack_name).item("nics").add($nic_name, $nic_exists)
+                            Clear-Variable -Name cur_nic
+                        }}
+                        elseif($res_type -like "*publicIpAddresses*"){{
+                            if(-not $list.item($stack_name).ContainsKey("pips")){{
+                                $list.item($stack_name).add("pips",@{{}})}}
+                            $pip_name = $res_type -split '/' | Select-Object -Last 1
+                            $cur_pip = Get-AzureRmPublicIpAddress -Name $pip_name `
+                            -ResourceGroupName \"{rg}\" -ErrorAction SilentlyContinue
+                            if(!$cur_pip){{$pip_exists = "false"}}
+                            else{{$pip_exists = "true"}}
+                            $list.Item($stack_name).item("pips").add($pip_name, $pip_exists)
+                            Clear-Variable -Name cur_pip
+                        }}
+                    }}
+                }}
+                $list | ConvertTo-Json -Depth 3
+            """.format(rg=self.resource_group, st_name=stack_name))
+
+        return ast.literal_eval(azure_data)
```

Comment:
@psav ast helped a lot before I made run_script return correct json. removing it